src/components/network/network-loader.js

```diff
@@ -19,117 +19,286 @@ genotet.utils.inherit(genotet.NetworkLoader, genotet.ViewLoader);
+ * @param {!Array<string>} genes Names of the genes.
  * @private
  */
-genotet.NetworkLoader.prototype.updateGeneRegex_ = function() {
-  var regex = '';
-  this.data.nodes.forEach(function(node, index) {
-    regex += node.id + (index == this.data.nodes.length - 1 ? '' : '|');
-  }, this);
-  this.data.geneRegex = regex;
+genotet.NetworkLoader.prototype.deleteGenes_ = function(genes) {
+  var geneMap = genotet.utils.keySet(genes);
+  // delete the nodes
+  for (var i = 0; i < this.data.network.nodes.length; i++) {
+    if (this.data.network.nodes[i].id in geneMap) {
+      this.data.network.nodes.splice(i, 1);
+    }
+  }
+  // delete the edges
+  for (var i = this.data.network.edges.length - 1; i >= 0; i--) {
+    if (this.data.network.edges[i].source in geneMap ||
+      this.data.network.edges[i].target in geneMap) {
+      this.data.network.edges.splice(i, 1);
+    }
+  }
 };
 
 /**
- * Fetches the incident edges of a given node.
- * @param {!Object} node Node of which the incident edges are queried.
+ * Adds one edge to the network.
+ * @param {!genotet.NetworkEdge} data Input data for adding an edge.
  */
-genotet.NetworkLoader.prototype.incidentEdges = function(node) {
-  var params = {
-    type: 'incident-edges',
-    fileName: this.data.fileName,
-    gene: node.id
-  };
-  this.get(genotet.data.serverURL, params, function(data) {
-    this.data.incidentEdges = data;
-    this.signal('incidentEdges');
-  }.bind(this), 'cannot get incident edges');
+genotet.NetworkLoader.prototype.addEdge = function(data) {
+  var source = data.source;
+  var target = data.target;
+  var weight = data.weight;
+  var sourceExists = false, targetExists = false;
+  this.data.network.nodes.forEach(function(node) {
+    if (source == node.id) {
+      sourceExists = true;
+    }
+    if (target == node.id) {
+      targetExists = true;
+    }
+  });
+  if (!sourceExists) {
+    this.data.network.nodes.push({
+      id: source,
+      label: source,
+      isTF: true
+    });
+  }
+  if (!targetExists) {
+    var isTF = false;
+    this.data.networkInfo.nodes.forEach(function(node) {
+      if (target == node.id) {
+        isTF = node.isTF;
+      }
+    });
+    this.data.network.nodes.push({
+      id: target,
+      label: target,
+      isTF: isTF
+    });
+  }
+  this.data.network.edges.push({
+    id: source + ',' + target,
+    source: source,
+    target: target,
+    weight: weight
+  });
+};
+
+/**
+ * Deletes one edge from the network.
+ * @param {{
+ *   source: string,
+ *   target: string
+ * }} data Input edge to be deleted.
+ */
+genotet.NetworkLoader.prototype.deleteEdge = function(data) {
+  var source = data.source;
+  var target = data.target;
+  for (var i = 0; i < this.data.network.edges.length; i++) {
+    if (this.data.network.edges[i].source == source &&
+      this.data.network.edges[i].target == target) {
+      this.data.network.edges.splice(i, 1);
+      break;
+    }
+  }
+  var sourceExists = false, targetExists = false;
```

Comment:
A bit confusing naming here. "source" and "target" are usually genes. You may want to say "sourceHasEdges" or similar.